### PR TITLE
WIP: Build qemu disk image (qcow2)

### PR DESCRIPTION
**Pull request by bdabelow**
Creation of a qemu disk (qcow2) in the build process using libguestfs tools.

Might have issues when building on Mac: https://github.com/rancher/k3os/issues/237#issuecomment-613797895

### CODE REVIEW

File: images/70-qemu/archs [hunk 1]
```diff
@@ -0,0 +1 @@
+amd64
```
Comment (dweomer) [inline images/70-qemu/archs:1]: Missing `arm64` and it looks like it is supported for `libguestfs-tools` on `debian:stable`.

File: images/output/01-full/Dockerfile [hunk 2]
```diff
@@ -10,6 +10,10 @@ ARG REPO
 ARG TAG
 FROM ${REPO}/k3os-kernel:${TAG} as kernel
 
+ARG REPO
+ARG TAG
+FROM ${REPO}/k3os-qemu:${TAG} as qemu
```
Comment (dweomer) [inline images/output/01-full/Dockerfile:15]: Exclusion of `arn64` from the qemu part of the build is causing it to fail. See https://github.com/rancher/k3os/pull/434/files#r410665601

File: install.sh [hunk 3]
```diff
@@ -269,6 +269,8 @@ create_opt()
     mkdir -p "${TARGET}/k3os/data/opt"
 }
 
+return 0 2>/dev/null || true
```
Comment (dweomer) [inline install.sh:272]: An excellent trick.

**Comment by dweomer** [comment]
Building this on my workstation and it hung with dapper trying to export the build and dist directories.

File: images/70-qemu/create-qemu-disk.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+add-drive "temp.img"
+
+set-network true
+
+launch
+
+part-disk /dev/sdb mbr
+mkfs ext4 /dev/sdb1
+mount /dev/sdb1 /
+tar-in /alpine-minirootfs-3.11.5-x86_64.tar.gz / compress:gzip
```
Comment (dweomer) [inline images/70-qemu/create-qemu-disk.sh:19]: Why do we need this?

Comment (bdabelow) [reply images/70-qemu/create-qemu-disk.sh:19]: In my initial approach as we are not booting from the ISO, getting the alpine mini rootfs was simple than unsquashing and mounting the k3os rootfs.

I am working on a revision that avoids libguestfs altogether and directly boots a patched ISO in qemu.

File: install.sh [hunk 5]
```diff
@@ -229,8 +260,13 @@ get_iso()
 
 setup_style()
 {
-    if [ "$K3OS_INSTALL_FORCE_EFI" = "true" ] || [ -e /sys/firmware/efi ]; then
+    if grep -q 'k3os.install.efi_hybrid=true' /proc/cmdline ; then
```
Comment (bdabelow) [inline install.sh:263]: @dweomer What is your opinion of the hybrid GPT/MBR added to install.sh? Still not sure whether it is worth the hassle.

Comment (dweomer) [reply install.sh:263]: @bdabelow what does it solve for?

Comment (bdabelow) [reply install.sh:263]: It allows creation of installations/disk images that will boot in BIOS/MBR as well as UEFI/GPT (virtual) machines.

**Comment by bdabelow** [comment]
Currently do not have timne to pursue this any further.

**Pull request closed without merging**